### DISCUSSION TITLE
Derby Stallion 96 can't dump properly.

### DISCUSSION THREAD

**Discussion by fakkuyuu** [sanni/cartreader#760] [Need help]
I have cleaned the terminals. I get the same CRC no matter how many times I try.

OSCR HW3 V12.4

[+] SNES/SFC (CLK0+1)

[+] SNES/SFC cartridge

Searching database...
Checksum: CC86
Header CRC32: 20B0E700
Found
Correcting size
Size: 32 -> 24Mbit
Banks: 128 -> 96

Title: DERBY STALLION 96
Revision: 0
Type: LoROM FastROM
ICs: ROM SAVE
Rom Size: 3 MB (96 banks)
Save Size: 32 KB
Checksum: CC86
Press Button...

[+] Read ROM

Saving to SNES/ROM/DERBY STALLION 96/1249/...
[*******************]
Checksum... 2BFE != CC86
Invalid Checksum
CRC32... CF963839 -> Not found

**Comment by nsx0r** [comment]
I have this game and I dumped it successfully a while ago on v9.1, I will try again with the latest version later.

edit:
cart serial = SHVC-ZDBJ-JPN
PCB serial =BSC-1A5M-02

```
OSCR HW5 V9.1

[+] Super Nintendo

[+] Super Nintendo


Searching database...

Name: DERBYSTALLION96
Type: LoROM FastROM
ICs: ROM SAVE
ROM Size: 32Mbit
Banks: 128 Chips: 2
Sram Size: 256Kbit
ROM Version: 1.0
Checksum: CC86
 
 

[+] Read Rom


Saving to SNES/ROM/DERBYSTALLION96/133/...
Internal Checksum...OK
CRC32... 19BDCB19 -> Derby Stallion 96 (Japan).sfc
```

**Reply by fakkuyuu** [reply]
I downgraded to 9.1 and tried again, same result
OSCR HW3 V9.1


Searching database...
Found
Correcting size

Name: DERBYSTALLION96
Type: LoROM FastROM
ICs: ROM SAVE
Rom Size: 24Mbit
Banks: 96 Chips: 2
Sram Size: 256Kbit
ROM Version: 1.0
Checksum: CC86
 
 


Saving to SNES/ROM/DERBYSTALLION96/1273/...
[*******************]
Internal Checksum...Checksum Error: 2BFE

CRC32... CF963839 -> Not found

**Reply by PsyK0p4T** [reply]
It seems you don't obtain the same rom size... nsx log shows 32mbits, and 24 mbits for fakkuyuu

**Reply by fakkuyuu** [reply]
When I compared it with a normal ROM using the bz editor, It seems that there is no data after 2MB.A screwdriver that can disassemble the cartridge is missing in the room. It takes time to prepare the Super Famicom main unit, so I can't check the operation.

**Reply by splash5** [reply]
I got the same result as @fakkuyuu with two "Derby Stallion 96". Force cart type to "4MB LoROM 256K SRAM" then it can be dumped properly, so as @nsx0r and @PsyK0p4T said, ROM size should be 32Mbits.
```
OSCR HW5 V12.5

[+] Super Nintendo/SFC

[+] SNES/SFC cartridge


Searching database...
Checksum: CC86
Header CRC32: 20B0E700
Found
Correcting size
Size: 32 -> 24Mbit
Banks: 128 -> 96

Title: DERBY STALLION 96
Revision: 0
Type: LoROM FastROM
ICs: ROM SAVE
Rom Size: 3 MB (96 banks)
Save Size: 32 KB
Checksum: CC86
Press Button...

[+] Read ROM


Saving to SNES/ROM/DERBY STALLION 96/898/...
[*******************]
Checksum... 2BFE != CC86
Invalid Checksum
CRC32... CF963839 -> Not found
Press Button...

[+] Force cart type

[+] 4MB LoROM 256K SRAM

Press Button...

[+] Read ROM


Saving to SNES/ROM/LoROM/899/...
Checksum... CC86 -> OK
CRC32... 19BDCB19 -> Derby Stallion 96 (Japan).sfc
Press Button...
```

**Reply by fakkuyuu** [reply]
thanks

OSCR HW3 V12.3

[+] SNES/SFC (CLK0+1)

[+] SNES/SFC cartridge


Searching database...
Checksum: CC86
Header CRC32: 20B0E700
Found
Correcting size
Size: 32 -> 24Mbit
Banks: 128 -> 96

Title: DERBY STALLION 96
Revision: 0
Type: LoROM FastROM
ICs: ROM SAVE
Rom Size: 3 MB (96 banks)
Save Size: 32 KB
Checksum: CC86
Press Button...

[+] Force cart type

[+] 4MB LoROM 256K SRAM

Press Button...

[+] Read ROM


Saving to SNES/ROM/LoROM/1297/...
Checksum... CC86 -> OK
CRC32... 19BDCB19 -> Derby Stallion 96 (Japan).sfc

**Comment by sanni** [comment]
In SNES.ino line [1298](https://github.com/sanni/cartreader/blob/master/Cart_Reader/SNES.ino#L1298)

Can you change:
````
if ((romType == LO) && (numBanks == 128) && (strcmp("CC86", checksumStr) == 0)) {
````

To: 
````
if ((romType == LO) && (numBanks == 96) && (strcmp("CC86", checksumStr) == 0)) {
````

And see if that fixes the issue?

**Reply by fakkuyuu** [reply]
When I tried as you said, I was able to dump normally.

OSCR HW3 V12.4

[+] SNES/SFC (CLK0+1)

[+] SNES/SFC cartridge


Searching database...
Checksum: CC86
Header CRC32: 20B0E700
Found
Correcting size
Size: 32 -> 24Mbit
Banks: 128 -> 96

Title: DERBY STALLION 96
Revision: 0
Type: LoROM FastROM
ICs: ROM SAVE
Rom Size: 3 MB (96 banks)
Save Size: 32 KB
Checksum: CC86
Press Button...

[+] Read ROM


Saving to SNES/ROM/DERBY STALLION 96/1370/...
Checksum... CC86 -> OK
CRC32... 19BDCB19 -> Derby Stallion 96 (Japan).sfc

**Comment by Prominos** [comment]
Confused reader here, does forcing the cart to 4MB imply trimming the resulting file to 3MB to get the correct content? Also I was wondering how is 24Mbit possible on a single chip? I thought it needed to be a power of two like 2, 4, 8, 16, 32, 64, 128, 256, 512 etc etc. I know it is 3MB but I'm having a hard time understanding how it works. Is it possible that the internal on-chip memory array only contains 24M cells and any addressing done at 0x300000 and above returns 00 or FF?
